app/controllers/api/v1/account_controller.rb

```diff
@@ -0,0 +1,20 @@
+# RESTful JSON API to retrieve data for My Account
+class Api::V1::AccountController < ApiController
+  before_action :authenticate_user!
+
+  def my_applications
+    applications = current_user_applications
+    listing_ids = applications.collect { |a| a['listingID'] }.uniq
+    listings = ListingService.listings(listing_ids.join(','))
+    applications.each do |app|
+      app['listing'] = listings.find { |l| l['listingID'] == app['listingID'] }
+    end
+    render json: { applications: applications }
+  end
```

Comment:
so for this action, we're getting all the applications of the current user, 
then we're getting the listing IDs of those applications,
then we're getting the listing objects of those listings and mapping them into the each corresponding application, so that each application has all the listing details.

very cool! wondering if we can make this a private method like `map_listing_data_to_applications`.